src/AppBundle/Service/OccurrenceSplitterService.php

```diff
@@ -0,0 +1,179 @@
+ *
+ * (c) 2017–2018 ITK Development
+ *
+ * This source file is subject to the MIT license.
+ */
+
+namespace AppBundle\Service;
+
+use AppBundle\Entity\DailyOccurrence;
+use AppBundle\Entity\Occurrence;
+use AppBundle\Entity\OccurrenceTrait;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyInfo\Extractor\ReflectionExtractor;
+
+class OccurrenceSplitterService
+{
+    private $dateSeparatorTimezone;
+    private $splitHour;
+    private $splitMinute;
+    private $splitSecond;
+
+    private $occurrenceTraitProperties;
+    private $propertyAccessor;
+
+    /**
+     * OccurrenceSplitterService constructor.
+     *
+     * @param string $dateSeparatorTime
+     * @param string $dateSeparatorTimezone
+     */
+    public function __construct(string $dateSeparatorTime, string $dateSeparatorTimezone)
+    {
+        $exploded = explode(':', $dateSeparatorTime);
+        $this->splitHour = (int) $exploded[0];
+        $this->splitMinute = (int) $exploded[1];
+        $this->splitSecond = (int) $exploded[2];
+        $this->dateSeparatorTimezone = new \DateTimeZone($dateSeparatorTimezone);
+
+        $propertyInfo = new ReflectionExtractor();
+        $this->occurrenceTraitProperties = $propertyInfo->getProperties(OccurrenceTrait::class);
+
+        $this->propertyAccessor = PropertyAccess::createPropertyAccessor();
+    }
+
+    /**
+     * Get new DailyOccurrences from an Occurrence.
+     *
+     * @param Occurrence $occurrence
+     *
+     * @throws \Exception
+     *
+     * @return Collection
+     */
+    public function getDailyOccurrences(Occurrence $occurrence): Collection
+    {
+        $dailyOccurrences = new ArrayCollection();
+        if ($occurrence->getStartDate() && $occurrence->getEndDate() && $occurrence->getStartDate() <= $occurrence->getEndDate()) {
+            $this->createDailyOccurrenceCollection($occurrence, $dailyOccurrences);
+        }
+
+        return $dailyOccurrences;
+    }
+
+    /**
+     * Copy values of OccurrenceTrait properties from one DailyOccurrence to another.
+     *
+     * @param DailyOccurrence $to
+     * @param DailyOccurrence $from
+     */
+    public function copyOccurrenceTraitPropertyValues(DailyOccurrence $to, DailyOccurrence $from): void
+    {
+        foreach ($this->occurrenceTraitProperties as $propertyPath) {
+            $value = $this->propertyAccessor->getValue($from, $propertyPath);
+            $this->propertyAccessor->setValue($to, $propertyPath, $value);
+        }
+    }
+
+    /**
+     * Create a collection of DailyOccurrences from an Occurrence.
+     *
+     * @param Occurrence $occurrence
+     * @param Collection $dailyOccurrences
+     *
+     * @throws \Exception
+     *
+     * @return Collection
+     */
+    private function createDailyOccurrenceCollection(Occurrence $occurrence, Collection $dailyOccurrences): Collection
```

Comment:
Why are we passing a `Collection` to `createDailyOccurrenceCollection` (it seems to be always empty)? Why not let the function create the collection?